### PR TITLE
Implement sort-strength and sort-weakness features

### LINKED ISSUE #102: As an organized and data-driven coach I can sort my players by strengths and weaknesses so I can get a better understanding of how players compare
Implement sort-strength and sort-weakness function.

Sort the list of persons by strengths and weaknesses in descending order respectively

Respective formats: sort-strength
                                sort-weakness

### PR DESCRIPTION
**Pull request by AAlghrairy**
1. Closes #102 
2. Implements the `sort-strength` and `sort-weakness` commands
3. Includes relevant test cases

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/SortWeaknessCommand.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        @Override
+        public int compare(Person o1, Person o2) {
+            //if total weaknesses are equal, order by least strengths
+            if (o1.getWeaknesses().size() == o2.getWeaknesses().size()) {
+                return o1.getStrengths().size() - o2.getStrengths().size();
+            } else if (o1.getWeaknesses().size() > o2.getWeaknesses().size()) {
+                return -1;
+            } else {
+                return 1;
+            }
```
Comment (juayhee) [inline src/main/java/seedu/address/logic/commands/SortWeaknessCommand.java:36]: Probably can be cleaned up to
`else {
    return o2.getWeaknesses().size().compareTo(o1.getWeaknesses().size())
}`

File: src/test/java/seedu/address/logic/commands/SortStrengthCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for SortStrengthCommand.
+ */
+public class SortStrengthCommandTest {
+
+    private Model model;
+    private Model expectedModel;
+
+    @BeforeEach
+    public void setUp() {
```
Comment (juayhee) [inline src/test/java/seedu/address/logic/commands/SortStrengthCommandTest.java:25]: Could be renamed to something that fits the common style of X_Y_Z.

File: src/test/java/seedu/address/logic/commands/SortWeaknessCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for SortWeaknessCommand.
+ */
+public class SortWeaknessCommandTest {
+
+    private Model model;
+    private Model expectedModel;
+
+    @BeforeEach
+    public void setUp() {
```
Comment (juayhee) [inline src/test/java/seedu/address/logic/commands/SortWeaknessCommandTest.java:25]: Could be renamed to something that fits the common style of X_Y_Z.

File: src/test/java/seedu/address/logic/commands/SortStrengthCommandTest.java (hunk 2 above)
Comment (AAlghrairy) [reply src/test/java/seedu/address/logic/commands/SortStrengthCommandTest.java:25]: The rationale for this name is that it is not a test method, but rather an initialization method (hence the `@BeforeEach` annotation).

File: src/main/java/seedu/address/logic/commands/SortWeaknessCommand.java (hunk 1 above)
Comment (AAlghrairy) [reply src/main/java/seedu/address/logic/commands/SortWeaknessCommand.java:36]: I kept the method in this format for the sake of readability. If the benefit doesn't outweigh the drawback, I'll consider reformatting these comparators and adding more comments. Thanks for bringing this up 👍

**Pull request merged**